pkg/verifier/safe_delete.go

```diff
@@ -31,23 +36,45 @@ func SafeDelete(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bac
 		return errors.Errorf("%s dir seems to exists in backup bucket. Remove this block manually if you are sure it is safe to do", id)
 	}
 
-	tempdir, err := ioutil.TempDir("", "safe-delete")
-	if err != nil {
-		return err
+	if tempdir == "" {
```

Comment:
Maybe I wasn't clear, I would expect that when `useExisting` is set, that `tempDir` must be set, so I would expect an error from this function if that is not the case, and this if statement to be combined with the parts below, I think that'll also make the code a lot more easy to understand.

Reply:
Ok, refactored this a bit.  It will return an error if you ask it to use an existing tempdir and do not provide one, and the logic is a little simpler.